Flaky TestServerShouldCloseWhenAllRequestsAreFulfilled

```
--- FAIL: TestServerShouldCloseWhenAllRequestsAreFulfilled (0.02s)
    backendrecorder_test.go:28: number of requests served does not match with the expected. Expected 4 but got 0
    backendrecorder_test.go:18: Get "http://127.0.0.1:46517/2": read tcp 127.0.0.1:54354->127.0.0.1:46517: read: connection reset by peer
FAIL
FAIL	github.com/zalando/skipper/proxy/backendtest	0.065s
```

--- a/proxy/backendtest/backendrecorder_test.go
+++ b/proxy/backendtest/backendrecorder_test.go
@@ -3,28 +3,33 @@ package backendtest
 import (
 	"io"
 	"net/http"
-	"strconv"
 	"testing"
 	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sync/errgroup"
 )
 
 func TestServerShouldCloseWhenAllRequestsAreFulfilled(t *testing.T) {
 	expectedRequests := 4
 	recorder := NewBackendRecorder(10 * time.Millisecond)
+
+	var g errgroup.Group
 	for i := 0; i < expectedRequests; i++ {
-		go func(counter int) {
-			resp, err := http.Get(recorder.GetURL() + "/" + strconv.Itoa(counter))
+		g.Go(func() error {
+			resp, err := http.Get(recorder.GetURL())
 			if err != nil {
-				t.Error(err)
-				return
+				return err
 			}
-			io.Copy(io.Discard, resp.Body)
-			resp.Body.Close()
-		}(i)
+			if _, err := io.Copy(io.Discard, resp.Body); err != nil {
+				return err
+			}
+			return resp.Body.Close()
+		})
 	}
+	require.NoError(t, g.Wait())
 	recorder.Done()
-	servedRequests := len(recorder.GetRequests())
-	if servedRequests != expectedRequests {
-		t.Errorf("number of requests served does not match with the expected. Expected %d but got %d", expectedRequests, servedRequests)
-	}
+
+	assert.Equal(t, expectedRequests, len(recorder.GetRequests()))
 }

